pkg/rpcclient/wsclient.go

```diff
@@ -831,6 +885,25 @@ func (c *WSClient) ReceiveNotaryRequests(flt *neorpc.TxFilter, rcvr chan<- *resu
 	return c.performSubscription(params, r)
 }
 
+// ReceivePersistedHeader registers provided channel as a receiver for new header
+// events. Events can be filtered by the given HeaderFilter, nil value doesn't add
+// any filter. See WSClient comments for generic Receive* behaviour details.
+func (c *WSClient) ReceivePersistedHeader(flt *neorpc.BlockFilter, rcvr chan<- *block.Header) (string, error) {
+	if rcvr == nil {
+		return "", ErrNilNotificationReceiver
+	}
+	params := []any{"header_of_added_block"}
```

Comment:
Add a check for client initialization, ref. https://github.com/nspcc-dev/neo-go/pull/3257.